freud/util.pyx

```diff
@@ -209,15 +210,25 @@ def _convert_array(array, shape=None, dtype=np.float32):
         dtype: :code:`dtype` to convert the array to if :code:`array.dtype`
             is different. If :code:`None`, :code:`dtype` will not be changed
             (Default value = :class:`numpy.float32`).
+        requirements (Sequence[str]): A sequence of string flags to be passed to
+            :func:`numpy.require`.
+        allow_copy (bool): If :code:`False` and the input array does not already
+            conform to the required dtype and other requirements, this function
+            will raise an error rather than coercing the array into a copy that
+            does satisfy the requirements (Default value = :code:`True`).
 
     Returns:
         :class:`numpy.ndarray`: Array.
     """
     if array is None:
         return np.empty(shape, dtype=dtype)
 
-    array = np.asarray(array)
-    return_arr = np.require(array, dtype=dtype, requirements=['C'])
+    return_arr = np.require(array, dtype=dtype, requirements=requirements)
```

Comment:
Could you explain why we don't want ```array = np.asarray(array)``` here? Is this for reducing the number of copies? I guess this is what's causing the errors

Reply:
It's redundant. `np.require` internally has to convert `array` to a numpy array anyway. The problems is that I didn't change enough of the code below, which is still using `array` instead of `return_arr`.